app/views/unattended/provisioning_templates/registration/remote_execution_pull_setup.erb

```diff
@@ -54,34 +54,45 @@ if [ -f $CONFIGTOML ]; then
     cat <<EOF > $CONFIGTOML
 # yggdrasil global configuration settings written by katello-pull-transport-migrate
 broker = ["mqtts://$SERVER_NAME:1883"]
+server = ["mqtts://$SERVER_NAME:1883"]
 cert-file = "$CERT_FILE"
 key-file = "$KEY_FILE"
 ca-root = ["$CA_FILE"]
 log-level = "error"
+data-host = ""
+path-prefix = "yggdrasil"
 EOF
 
 else
     echo "$SYSCONFDIR/yggdrasil/config.toml not found! Did 'yum install yggdrasil' succeed?"
     exit 1
 fi
 
+YGGDRASIL_SERVICE=yggdrasil.service
+systemctl cat $YGGDRASIL_SERVICE >/dev/null 2>/dev/null || YGGDRASIL_SERVICE=yggdrasild.service
+
 # start the yggdrasild service
-if systemctl is-enabled yggdrasild 2>/dev/null && [ -n "$YGGDRASIL_RESTART_DELAY" ] && [ "$YGGDRASIL_RESTART_DELAY" -gt 0 ]; then
-    systemd-run --on-active="$YGGDRASIL_RESTART_DELAY" --collect --no-block systemctl restart yggdrasild
+if systemctl is-enabled $YGGDRASIL_SERVICE 2>/dev/null && [ -n "$YGGDRASIL_RESTART_DELAY" ] && [ "$YGGDRASIL_RESTART_DELAY" -gt 0 ]; then
+    systemd-run --on-active="$YGGDRASIL_RESTART_DELAY" --collect --no-block systemctl restart $YGGDRASIL_SERVICE
 else
-    echo "Starting yggdrasild..."
-    systemctl try-restart yggdrasild
-    systemctl enable --now yggdrasild
+    echo "Starting $YGGDRASIL_SERVICE..."
+    systemctl try-restart $YGGDRASIL_SERVICE
+    systemctl enable --now $YGGDRASIL_SERVICE
 
-    # check status of yggdrasild and fail if it is not running
+    # check status of $YGGDRASIL_SERVICE and fail if it is not running
     # possible failure reason: incorrect protocol (should be tcp:// or mqtt://) or port (should be 1883)
     # also, cert-file and key-file must be valid
     # and broker must be running on the server
-    yggdrasil status
+    # yggdrasil-0.2.z provides a dedicated status subcommand, newer yggdrasil doesn't have anything like that
+    if which yggdrasil >/dev/null 2>/dev/null; then
+        yggdrasil status
+    else
+        systemctl status $YGGDRASIL_SERVICE
```

Comment:
the registration process will stop on this line as the systemctl status will occupy the shell, I think something like `systemctl status $YGGDRASIL_SERVICE --no-pager` should do it

Reply:
Or we could just not do it at all? I'd say it is just eyecandy, we still check if the service is active a couple of lines lower

Reply:
fine with me :)

Reply:
Dropped